tests/e2e/p/interchain_workflow.go

```diff
@@ -0,0 +1,220 @@
+					End:    uint64(endTime.Unix()),
+					Wght:   weight,
+				}},
+				nodePOP,
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{rewardKey.Address()},
+				},
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{rewardKey.Address()},
+				},
+				delegationFee,
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("adding a delegator to the new node", func() {
+			startTime := time.Now().Add(e2e.DefaultValidatorStartTimeDiff)
+			// Delegation duration doesn't actually matter to this
+			// test - it is only ensuring that adding a delegator
+			// doesn't break interchain transfer.
+			endTime := startTime.Add(15 * time.Second)
+
+			rewardKey, err := factory.NewPrivateKey()
+			require.NoError(err)
+
+			_, err = pWallet.IssueAddPermissionlessDelegatorTx(
+				&txs.SubnetValidator{Validator: txs.Validator{
+					NodeID: nodeID,
+					Start:  uint64(startTime.Unix()),
+					End:    uint64(endTime.Unix()),
+					Wght:   weight,
+				}},
+				pWallet.AVAXAssetID(),
+				&secp256k1fx.OutputOwners{
+					Threshold: 1,
+					Addrs:     []ids.ShortID{rewardKey.Address()},
+				},
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("exporting AVAX from the P-Chain to the X-Chain", func() {
+			_, err := pWallet.IssueExportTx(
+				xWallet.BlockchainID(),
+				exportOutputs,
+				e2e.WithDefaultContext(),
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("importing AVAX from the P-Chain to the X-Chain", func() {
+			_, err := xWallet.IssueImportTx(
+				constants.PlatformChainID,
+				&recipientOwner,
+				e2e.WithDefaultContext(),
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("checking that the recipient address has received imported funds on the X-Chain", func() {
+			balances, err := xWallet.Builder().GetFTBalance(common.WithCustomAddresses(set.Of(
+				recipientKey.Address(),
+			)))
+			require.NoError(err)
+			require.Positive(balances[avaxAssetID])
+		})
+
+		ginkgo.By("exporting AVAX from the P-Chain to the C-Chain", func() {
+			_, err := pWallet.IssueExportTx(
+				cWallet.BlockchainID(),
+				exportOutputs,
+				e2e.WithDefaultContext(),
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("initializing a new eth client")
+		ethClient := e2e.Env.NewEthClient(nodeURI)
+
+		ginkgo.By("importing AVAX from the P-Chain to the C-Chain", func() {
+			_, err := cWallet.IssueImportTx(
+				constants.PlatformChainID,
+				recipientEthAddress,
+				e2e.WithDefaultContext(),
+				e2e.WithSuggestedGasPrice(ethClient),
+			)
+			require.NoError(err)
+		})
+
+		ginkgo.By("checking that the recipient address has received imported funds on the C-Chain")
+		e2e.Eventually(func() bool {
+			balance, err := ethClient.BalanceAt(e2e.DefaultContext(), recipientEthAddress, nil)
+			require.NoError(err)
+			return balance.Cmp(big.NewInt(0)) > 0
+		}, e2e.DefaultTimeout, e2e.DefaultPollingInterval, "failed to see recipient address funded before timeout")
```

Comment:
Because this `ethClient` is talking to the same node that the `cWallet` is talking to - do we need this `Eventually`? I would have thought that this should always pass first try.